src/aaudio/AudioStreamAAudio.cpp

```diff
@@ -221,14 +221,17 @@ Result AudioStreamAAudio::open() {
     return result;
 }
 
-Result AudioStreamAAudio::close()
-{
+Result AudioStreamAAudio::close() {
     // The main reason we have this mutex if to prevent a collision between a call
     // by the application to stop a stream at the same time that an onError callback
     // is being executed because of a disconnect. The close will delete the stream,
     // which could otherwise cause the requestStop() to crash.
     std::lock_guard<std::mutex> lock(mLock);
 
+    // Update frame counter variables to avoid retrograde motion.
```

Comment:
You're updating something by calling `get` methods which feels weird. Could you use a setter instead or even `incrementFramesRead(0)` at a push?

Reply:
Right. That does feel wierd. I will add an update method.